velox/exec/HashAggregation.cpp

```diff
@@ -119,8 +119,9 @@ HashAggregation::HashAggregation(
     constantLists.push_back(constants);
   }
 
-  // Check that aggregate result type match the output type
+  // Initialize and check that aggregate result type match the output type
   for (auto i = 0; i < aggregates.size(); i++) {
+    aggregates[i]->initialize(driverCtx->queryConfig());
```

Comment:
Would it make sense to pass config to Aggregate::create and have it pass to AggregateFunctionFactory so that config is provided at the time when function is created and not after?

Reply:
I try as you suggest, but it will change the code too lot, every file in folder `velox/functions/prestosql/aggregates/` and ``velox/functions/sparksql/aggregates/` will be affected, and we cannot easily get QueryConfig in https://github.com/facebookincubator/velox/blob/main/velox/exec/AggregateCompanionAdapter.cpp#L440 'registerExtractFunctionWithSuffix' unless we add QueryConfig to `VectorFunctionFactory`,  or just an empty config.

Can you check this issue?

Reply:
CC: @kagamiori Wei, would you help look into this?

Reply:
@kagamiori Do you have some idea?

Reply:
Do you have some idea? @mbasmanova

Reply:
@jinchengchenghh I think it would be cleaner to pass config to Aggregate::create and getVectorFunction. It would also make it consistent with SimpleFunctionAdapterFactory::createVectorFunction. I do appreciate that would require some refactoring and changing many files, but I think that's Ok.